x-pack/plugins/security_solution/server/lib/detection_engine/notifications/legacy_rules_notification_alert_type.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  DEFAULT_RULE_NOTIFICATION_QUERY_SIZE,
+  LEGACY_NOTIFICATIONS_ID,
+  SERVER_APP_ID,
+} from '../../../../common/constants';
+
+// eslint-disable-next-line no-restricted-imports
+import { LegacyNotificationAlertTypeDefinition } from './legacy_types';
+import { AlertAttributes } from '../signals/types';
+import { siemRuleActionGroups } from '../signals/siem_rule_action_groups';
+import { scheduleNotificationActions } from './schedule_notification_actions';
+import { getNotificationResultsLink } from './utils';
+import { getSignals } from './get_signals';
+
+/**
+ * @deprecated Once legacy notifications/"side car actions" goes away this should also be removed
+ */
+export const legacyRulesNotificationAlertType = ({
+  logger,
+}: {
+  logger: Logger;
+}): LegacyNotificationAlertTypeDefinition => ({
+  id: LEGACY_NOTIFICATIONS_ID,
+  name: 'SIEM notification',
+  actionGroups: siemRuleActionGroups,
+  defaultActionGroupId: 'default',
+  producer: SERVER_APP_ID,
+  validate: {
+    params: schema.object({
+      ruleAlertId: schema.string(),
+    }),
+  },
+  minimumLicenseRequired: 'basic',
+  isExportable: false,
+  async executor({ startedAt, previousStartedAt, alertId, services, params }) {
+    const ruleAlertSavedObject = await services.savedObjectsClient.get<AlertAttributes>(
+      'alert',
+      params.ruleAlertId
+    );
+
+    if (!ruleAlertSavedObject.attributes.params) {
+      logger.error(`Saved object for alert ${params.ruleAlertId} was not found`);
+      return;
+    }
+
+    const { params: ruleAlertParams, name: ruleName } = ruleAlertSavedObject.attributes;
+    const ruleParams = { ...ruleAlertParams, name: ruleName, id: ruleAlertSavedObject.id };
+
+    const fromInMs = parseScheduleDates(
+      previousStartedAt
+        ? previousStartedAt.toISOString()
+        : `now-${ruleAlertSavedObject.attributes.schedule.interval}`
+    )?.format('x');
+    const toInMs = parseScheduleDates(startedAt.toISOString())?.format('x');
+
+    const results = await getSignals({
+      from: fromInMs,
+      to: toInMs,
+      size: DEFAULT_RULE_NOTIFICATION_QUERY_SIZE,
+      index: ruleParams.outputIndex,
+      ruleId: ruleParams.ruleId,
+      esClient: services.scopedClusterClient.asCurrentUser,
+    });
+
+    const signals = results.hits.hits.map((hit) => hit._source);
+
+    const signalsCount =
+      typeof results.hits.total === 'number' ? results.hits.total : results.hits.total.value;
+
+    const resultsLink = getNotificationResultsLink({
+      from: fromInMs,
+      to: toInMs,
+      id: ruleAlertSavedObject.id,
+      kibanaSiemAppUrl: (ruleAlertParams.meta as { kibana_siem_app_url?: string } | undefined)
+        ?.kibana_siem_app_url,
+    });
+
+    logger.info(
+      `Found ${signalsCount} signals using signal rule name: "${ruleParams.name}", id: "${params.ruleAlertId}", rule_id: "${ruleParams.ruleId}" in "${ruleParams.outputIndex}" index`
+    );
```

Comment:
From the call today can we add a "LEGACY" or some other signifier to this log line, and change this from `logger.info` to `logger.warn` with some text like `"This is deprecated. In the near future please re-save (?) your rules"`

Reply:
I fixed this up and added more messages to this.